lib/ui/src/provider.ts

```diff
@@ -1,9 +1,9 @@
 export default class Provider {
-  getElements() {
+  getElements(type: string) {
```

Comment:
fyi @ndelangen  I had to modify the signature of `getElements` and `handleAPI` because every class that has extended this base class had these arguments implemented

I hope that's ok?